tests/integration_tests/functional/test_cpu_features_aarch64.py

```diff
@@ -87,39 +108,47 @@ def test_default_cpu_features(test_microvm_with_api):
     vm.basic_config()
     vm.add_net_iface()
     vm.start()
-    _check_cpu_features_arm(vm)
+    _check_cpu_features_arm(vm, "4.14")
 
 
 @pytest.mark.skipif(
     PLATFORM != "aarch64",
     reason="This is aarch64 specific test.",
 )
 @nonci_on_arm
-def test_cpu_features_with_static_template(test_microvm_with_api, cpu_template):
+def test_cpu_features_with_static_template(
+    microvm_factory, guest_kernel, rootfs_ubuntu_22, cpu_template
+):
     """
     Check the CPU features for a microvm with the specified config.
     """
-    vm = test_microvm_with_api
+
+    vm = microvm_factory.build(guest_kernel, rootfs_ubuntu_22, monitor_memory=False)
     vm.spawn()
     vm.basic_config(cpu_template=cpu_template)
     vm.add_net_iface()
     vm.start()
-    _check_cpu_features_arm(vm, "aarch64_v1n1")
+    guest_kv = re.search("vmlinux-(.*).bin", guest_kernel.name()).group(1)
```

Comment:
Maybe we should extract this as a method of the Microvm class. For example:

```
uvm.get_kernel_version_str()
(5, 1)
```

Reply:
I can do a follow up PR with this change